docs/e2e/requirements.md

```diff
@@ -0,0 +1,53 @@
+#### Supported test focuses
+
+The following are ginkgo test focuses supported by openshift on azure e2e tests
+
+* `[EveryPR]` - These tests should be run on every PR
+* `[LongRunning]` - These tests are long running and should be run periodically
+* `[Real]` - These tests should be run against the real RP
+* `[Fake]` - These tests should be run against the fake RP
+
+All tests require the specification of either a [Real] or [Fake] focus. 
+
+#### Running the e2e tests with focuses
+
+You can specify a test focus by setting the `FOCUS` environment variable. For example
+to run all tests tagged with [EveryPR] against the Real RP you would specify a focus as
+
+```
+export FOCUS=\[EveryPR\]\[Real\]
+./hack/e2e.sh
+```
+
+Note: even if this document refers to focuses such as [Real], you should always escape them
+before passing them to `FOCUS`.
+
+#### Real RP environment requirements
+
+The following are required for running the e2e tests against the real RP
+
+| Artifact Kind | Name | Notes |
+|--- | --- | --- |
+| `Ginkgo Focus` | `Ginkgo Focus` | `[Real]` | Allows the test to configure its `OpenshiftManagedCluster` client against the real RP |
+| `Environment Variable` | `AZURE_CLIENT_ID` | Required for authentication against the real RP |
+| `Environment Variable` | `AZURE_CLIENT_SECRET` | Required for authentication against the real RP |
+| `Environment Variable` | `AZURE_TENANT_ID` | Required for authentication against the real RP |
+| `Environment Variable` | `AZURE_SUBSCRIPTION_ID` | The subscription id for an existing cluster |
+| `Environment Variable` | `AZURE_REGION` | The region for an existing cluster |
+| `Environment Variable` | `RESOURCEGROUP` | The resource group for an existing cluster |
+
+#### Fake RP environment requirements
+
+The following are required for running the e2e tests against the fake RP
+
+| Artifact Kind | Name | Notes |
+| --- | --- | --- |
+| `Ginkgo Focus` | `[Fake]` | Allows the test to configure its `OpenshiftManagedCluster` client against the fake RP |
+| `File` | `_data/containerservice.yaml` | Allows the test to retrieve credentials as well as to decide if it is a `create` vs `update` |
```

Comment:
Actually _data/manifest.yaml is also required (used by the reentrant update test now and by the key rotation test although it should be removed from the latter once key rotation is a supported action in the fake RP. It can potentially be removed from the former too maybe in favor of reading the manifest from containerservice.yaml), and test/manifests/normal/scaleup.yaml and test/manifests/normal/scaledown.yaml are used by the scaleupdown test. Can you provide examples on how to make the scale up/down manifests configurable since we have flags for them?

Reply:
@kargakis mostly took care of your suggestions. thanks. Regarding examples on how to make the scale up/down configurable through the use of flags, it should be possible to run the binary with the flags as shown below

```
./e2e -scaleUpManifest=/some/path/to/a/scale/down/manifest  (other flags)
```

I'll update the documentation in this PR with examples once #914 merges.